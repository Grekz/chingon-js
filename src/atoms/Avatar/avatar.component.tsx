import styled from '@emotion/styled'
import * as React from 'react'

interface IAvatarProps {
  content: string
  backgroundColor?: string
  hasImage?: boolean
  size?: string
}

type AvatarType = React.SFC<IAvatarProps>

const Avatar = (props: IAvatarProps) => (
  <AvatarWrapper {...props}>
    {props.hasImage ? null : <span>{props.content}</span>}
  </AvatarWrapper>
)
Avatar.defaultProps = {
  backgroundColor: 'none',
  size: '100px',
}

const AvatarWrapper: AvatarType = styled.div`
  display: inline-flex;
  border-radius: 50%;
  background-color: ${({ backgroundColor }) => backgroundColor};
  background-image: ${({ content, hasImage }) =>
    hasImage ? `url(${content})` : null};
  background-size: 100%;
  width: ${({ size }) => size};
  height: ${({ size }) => size};
  align-content: center;
  text-align: center;
  flex-flow: row wrap;
  vertical-align: top;
  > span {
    width: 100%;
    font-size: 2rem;
  }
`

export default Avatar
