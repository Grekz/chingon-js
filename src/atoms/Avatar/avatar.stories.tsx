import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import Avatar from '.'

storiesOf('Atoms|Avatar', module).add('As Default', () => (
  <>
    <Avatar content="AC" backgroundColor="lightgray" />
    <Avatar
      size="150px"
      content="https://avatars1.githubusercontent.com/u/3975603"
      hasImage={true}
    />
  </>
))
