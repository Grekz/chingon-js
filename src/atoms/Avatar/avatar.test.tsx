import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import Avatar from '.'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<Avatar /> rendering', () => {
  it('should render Avatar', () => {
    const wrapper = shallow(
      <Avatar
        size="150px"
        content="https://avatars1.githubusercontent.com/u/3975603"
        hasImage={true}
      />
    )
    expect(wrapper).toHaveLength(1)
  })
  it('should match snapshot', () => {
    const tree = shallow(
      <Avatar
        size="150px"
        content="https://avatars1.githubusercontent.com/u/3975603"
        hasImage={true}
      />
    )
    expect(toJson(tree)).toMatchSnapshot()
  })
})
