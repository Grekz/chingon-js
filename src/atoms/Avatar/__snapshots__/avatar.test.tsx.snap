// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> rendering should match snapshot 1`] = `
<Styled(div)
  backgroundColor="none"
  content="https://avatars1.githubusercontent.com/u/3975603"
  hasImage={true}
  size="150px"
/>
`;
